helmclient.go

```diff
@@ -67,13 +68,28 @@ func New(config Config) (*Client, error) {
 
 // DeleteRelease uninstalls a chart given its release name.
 func (c *Client) DeleteRelease(releaseName string, options ...helmclient.DeleteOption) error {
-	t, err := c.newTunnel()
-	if err != nil {
-		return microerror.Mask(err)
+	o := func() error {
```

Comment:
I wrapped everything in backoffs. Please check if the calls are right. The great unit testing we did here so far helped a lot sorting this out, especially with the error handling. Thank you for the good work so far.